core/src/main/java/io/github/wysohn/triggerreactor/core/script/interpreter/Interpreter.java

```diff
@@ -406,6 +406,185 @@ private void start(Node node, InterpreterLocalContext localContext) throws Inter
+                        }
+
+                        matches = true;
+                        start(caseBody, localContext);
+                        break;
+                    }
+                } else {
+                    for (int j = 0; j < parameters.getChildren().size(); j++) {
+                        final Node parameter = parameters.getChildren().get(j);
+                        start(parameter, localContext);
+
+                        if (localContext.isStopFlag()) {
+                            return;
+                        }
+
+                        final Token rawParameterToken = localContext.popToken();
+                        final Token parameterToken;
+                        // Smart casting for enum types, or do default conversions.
+                        if (variableNameToken.isEnum() && rawParameterToken.isString()) {
+                            Token maybeParameterToken;
+                            try {
+                                final Class<Enum> enumClass = (Class<Enum>) variableNameToken.value.getClass();
+                                maybeParameterToken = parseValue(
+                                    Enum.valueOf(enumClass, rawParameterToken.value.toString()),
+                                    rawParameterToken
+                                );
+                            } catch (final IllegalArgumentException ignored) {
+                                maybeParameterToken = tryUnwrapVariable(rawParameterToken, localContext);
+                            }
+
+                            parameterToken = maybeParameterToken;
+                        } else {
+                            parameterToken = tryUnwrapVariable(rawParameterToken, localContext);
+                        }
+
+                        if (variableType != Type.EPS && !variableType.equals(parameterToken.getType())) {
+                            throw new InterpreterException("Mismatched type for parameter " + rawParameterToken + "! Expected " + variableType + " but found " + parameterToken.getType());
+                        }
+
+                        if (variableNameToken.getValue().equals(parameterToken.getValue())) {
+                            final Node caseBody = caseNode.getChildren().get(1);
+                            if (!Type.CASEBODY.equals(caseBody.getToken().getType())) {
+                                throw new InterpreterException("Expected case body but found " + parameters);
+                            }
+
+                            matches = true;
+                            start(caseBody, localContext);
+                            break iterCaseOrDefaultNodes;
+                        }
+                    }
+                }
+            }
+
+            if (!matches) {
+                // TODO(Sayakie): Raises an exception or ignore silently?
+                throw new InterpreterException("No matched arm");
```

Comment:
I think it's okay to throw an exception here, given that we already make sure that there is at least one arm exist for the SWITCH in the parsing step